nixos/modules/services/networking/soju.nix

```diff
@@ -109,6 +109,17 @@ in
       default = "";
       description = "Lines added verbatim to the configuration file.";
     };
+
+    configFile = mkOption {
+      type = types.path;
+      default = configFile;
+      defaultText = "Config file generated from other options.";
+      description = ''
+        Path to config file. If this option is set, it will override any
+        configuration done using other options.
+      '';
+      example = literalExpression "./soju.conf";
```

Comment:
Relative paths are specifically not allowed.

Reply:
Why are relative paths not allowed? Since it is a literal expression, it will be added to nix store automatically. Options like `services.xserver.windowManager.qtile.configFile` also use relative paths as an example.

Reply:
Maybe I was a bit imprecise. `"/absolute/path/soju.conf"` is allowed. `./soju.conf` is allowed. `"./soju.conf"` is not. I guess it is okay as an example.